anna/my-app/src/components/searchForm/SearchForm.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import './searchForm.css';
+import Button from '../../components/button';
+
+import * as QueryString from "query-string";
+import { RouteComponentProps } from "react-router-dom";
+
+interface ISearchFormProps {
+  location: RouteComponentProps["location"];
+  onSubmit(arg0: ISearchFormState): void;
+  onSearchByChange(arg0: ISearchFormState): void;
+}
+
+export interface ISearchFormState {
+  search: string;
+  searchBy: SearchType;
+}
+
+export enum SearchType {
+  Title = "title",
+  Ganre = "genres"
+}
+
+class SearchForm extends React.Component<ISearchFormProps, ISearchFormState> {
+  state: ISearchFormState = {
+    search: '',
+    searchBy: SearchType.Title
+  };
+
+  changeValue = (e: React.ChangeEvent<HTMLInputElement>) => this.setState({ search: e.target.value });
+
+  handleSearchByChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      search: '',
+      searchBy: e.target.value as SearchType,
+    }, () => {
+      this.props.onSearchByChange(this.state);
+    });
+  }
+
+  componentDidMount() {
+    const { location } = this.props;
+    console.log(location.search, 'search mount')
+    const searchParams = QueryString.parse(location.search);
+
+    if (searchParams.searchBy) {
+      this.setState({
+        searchBy: searchParams.searchBy as SearchType
+      });
+    } 
+
+    if (searchParams.search) {
+      this.setState({
+        search: searchParams.search as SearchType
+      })
+    }
+  }
```

Comment:
1) Usually it's better to keep lifecycle methods on the top of the class. 
2) It is not really good approach to call setState several times in a row. Better to set all at once
```
this.setState({
        search: searchParams.search as SearchType || '',
        searchBy: searchParams.searchBy as SearchType || SearchType.Title
      })
```